scripts/genpinctrl/config.json

```diff
@@ -0,0 +1,219 @@
+            "match": "^SPI\\d+_MISO$",
+            "drive": "push-pull",
+            "bias": "pull-down",
+            "mode": "input",
+            "variant": "master"
+        },
+        {
+            "name": "SPI_MASTER_MOSI",
+            "match": "^SPI\\d+_MOSI$",
+            "drive": "push-pull",
+            "bias": "pull-down",
+            "mode": "output",
+            "variant": "master"
+        },
+        {
+            "name": "SPI_MASTER_SCK",
+            "match": "^SPI\\d+_SCK$",
+            "drive": "push-pull",
+            "bias": "pull-down",
+            "mode": "output",
+            "variant": "master"
+        },
+        {
+            "name": "SPI_MASTER_NSS",
+            "match": "^SPI\\d+_NSS$",
+            "drive": "push-pull",
+            "bias": "pull-up",
+            "mode": "output",
+            "variant": "master"
+        },
+        {
+            "name": "SPI_SLAVE_MISO",
+            "match": "^SPI\\d+_MISO$",
+            "drive": "push-pull",
+            "bias": "pull-down",
+            "mode": "output",
+            "variant": "slave"
+        },
+        {
+            "name": "SPI_SLAVE_MOSI",
+            "match": "^SPI\\d+_MOSI$",
+            "drive": "push-pull",
+            "bias": "pull-down",
+            "mode": "input",
+            "variant": "slave"
+        },
+        {
+            "name": "SPI_SLAVE_SCK",
+            "match": "^SPI\\d+_SCK$",
+            "drive": "push-pull",
+            "bias": "pull-down",
+            "mode": "input",
+            "variant": "slave"
+        },
+        {
+            "name": "SPI_SLAVE_NSS",
+            "match": "^SPI\\d+_NSS$",
+            "drive": "push-pull",
+            "bias": "pull-up",
+            "mode": "input",
+            "variant": "slave"
+        },
+        {
+            "name": "TIM_CH / TIM_CHN",
+            "match": "^TIM\\d+_CH\\d+N?$",
+            "drive": "push-pull",
+            "bias": "disable",
+            "mode": "output"
+        },
+        {
+            "name": "TIM_BKIN / TIM_BKIN2",
+            "match": "^TIM\\d+_BKIN\\d?$",
+            "bias": "pull-up",
+            "mode": "input"
+        },
+        {
+            "name": "TIM_ETR",
+            "match": "^TIM\\d+_ETR$",
+            "bias": "disable",
+            "mode": "input"
+        },
+        {
+            "name": "UART_CTS / USART_CTS / LPUART_CTS",
+            "match": "^(?:LP)?US?ART\\d+_CTS$",
+            "drive": "open-drain",
+            "bias": "pull-up",
+            "mode": "input"
+        },
+        {
+            "name": "UART_RTS / USART_RTS / LPUART_RTS",
+            "match": "^(?:LP)?US?ART\\d+_RTS$",
+            "drive": "open-drain",
+            "bias": "pull-up",
+            "mode": "output"
+        },
+        {
+            "name": "UART_TX / USART_TX / LPUART_TX",
+            "match": "^(?:LP)?US?ART\\d+_TX$",
+            "drive": "push-pull",
+            "bias": "pull-up",
```

Comment:
I think it's better to have a single configuration file, there'll be lots of duplicate content otherwise. Config file can be made more explicit for F1 only fields (e.g. `mode`) or non-F1 (e.g. `drive`). I still think it's better to keep things simple being a single family that diverges, it can be easily solved by an if/else in the Jinja template as it's already done for the include and pinmux entry.

Reply:
@erwango I have added conditional `bias` rendering depending on family and for F1, only render if `mode` is not `output` or `inout` (both map to `ALTERNATE` output). Can you check?

Reply:
The way I see it is that user should be able, in this(these) config file(s), to see/set the description that he can see in the ref manual and expect in the dts files. 
The whole idea is to have something easy to deal with from the reference manual to the final pin configuration (and the other way round). The dts itself is just a intermediate step used as a normalized description.
I tried to follow this approach when working on the bindings, this is why we end up 2 different -pinctrl bindings (while I could have kept the same description and tweak the driver to abstract differences).

I would expect the pinctrl generation script allows to follow the same approach and keep configuration transparent to the user. So each one can verify that what is available in the config is aligned with  the ref man and the IP settings. This is why I'd prefer having 2 different config files when it makes sense, rather that tweaking the values in the script, even at cost of some config duplication.

Reply:
> The way I see it is that user should be able, in this(these) config file(s), to see/set the description that he can see in the ref manual and expect in the dts files.
> The whole idea is to have something easy to deal with from the reference manual to the final pin configuration (and the other way round). The dts itself is just a intermediate step used as a normalized description.
> I tried to follow this approach when working on the bindings, this is why we end up 2 different -pinctrl bindings (while I could have kept the same description and tweak the driver to abstract differences).
>
If we want to have comments in the config file we can switch to YAML, as JSON does not allow comments. I can take a look if you want, should be easy.
 
> I would expect the pinctrl generation script allows to follow the same approach and keep configuration transparent to the user. So each one can verify that what is available in the config is aligned with the ref man and the IP settings. This is why I'd prefer having 2 different config files when it makes sense, rather that tweaking the values in the script, even at cost of some config duplication.

Ok, I understand the idea. Let me see what I can do.

Reply:
> > The way I see it is that user should be able, in this(these) config file(s), to see/set the description that he can see in the ref manual and expect in the dts files.
> > The whole idea is to have something easy to deal with from the reference manual to the final pin configuration (and the other way round). The dts itself is just a intermediate step used as a normalized description.
> > I tried to follow this approach when working on the bindings, this is why we end up 2 different -pinctrl bindings (while I could have kept the same description and tweak the driver to abstract differences).
> 
> If we want to have comments in the config file we can switch to YAML, as JSON does not allow comments. I can take a look if you want, should be easy. In fact, YAML may be more familiar to people because it's used in the DT bindings, we could also add the `description` field for detailed documentation.
 
The format itself sounds good to me (initially I don't have particular expectations on comments) but indeed yaml may be a good alternative.

> > I would expect the pinctrl generation script allows to follow the same approach and keep configuration transparent to the user. So each one can verify that what is available in the config is aligned with the ref man and the IP settings. This is why I'd prefer having 2 different config files when it makes sense, rather that tweaking the values in the script, even at cost of some config duplication.
> 
> Ok, I understand the idea. Let me see what I can do.

Great, txs